File: app/src/main/java/eu/kanade/tachiyomi/data/rest/GithubService.java
package eu.kanade.tachiyomi.data.rest;

import retrofit.http.GET;
import rx.Observable;


/**
 * Used to connect with the Github API
 */
public interface GithubService {
    String SERVICE_ENDPOINT = "https://api.github.com";

    @GET("/repos/inorichi/tachiyomi/releases/latest") Observable<Release> getLatestVersion();

}

File: app/src/main/java/eu/kanade/tachiyomi/data/updater/GithubService.kt
package eu.kanade.tachiyomi.data.updater

import eu.kanade.tachiyomi.network.NetworkHelper
import retrofit2.Retrofit
import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import rx.Observable
import uy.kohesive.injekt.Injekt
import uy.kohesive.injekt.api.get

/**
 * Used to connect with the Github API.
 */
interface GithubService {

    companion object {
        fun create(): GithubService {
            val restAdapter = Retrofit.Builder()
                    .baseUrl("https://api.github.com")
                    .addConverterFactory(GsonConverterFactory.create())
                    .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
                    .client(Injekt.get<NetworkHelper>().client)
                    .build()

            return restAdapter.create(GithubService::class.java)
        }
    }

    @GET("/repos/inorichi/tachiyomi/releases/latest")
    fun getLatestVersion(): Observable<GithubRelease>

}